tests/test_engine/test_engine.py

```diff
@@ -8,78 +8,52 @@
-    for model_name in test_models:
-        get_components_func = non_distributed_component_funcs.get_callable(model_name)
-        model_builder, train_dataloader, _, optimizer_class, criterion = get_components_func()
-
-        model = model_builder(checkpoint=False)
-        engine, train_dataloader, *args = colossalai.initialize(model=model,
-                                                                optimizer=optimizer_class(model.parameters(), lr=1e-3),
-                                                                criterion=criterion,
-                                                                train_dataloader=train_dataloader)
-
-        try:
-            engine.train()
-            for data, label in train_dataloader:
-                engine.zero_grad()
-                data = data.cuda()
-                label = label.cuda()
-                if criterion:
-                    output = engine(data)
-                    loss = engine.criterion(output, label)
-                else:
-                    loss = engine(data, label)
-                engine.backward(loss)
-                engine.step()
-                break
-        except IndexError:
-            # if using apex amp, NetWithRepeatedlyComputedLayers will raise an index out of range issue
-            # the following check fails in apex
-            # if cached_x.grad_fn.next_functions[1][0].variable is not x:
-            continue
-
-
-def run_with_no_amp():
-    run_train()
-
-
-def run_with_torch_amp():
-    # hack config
-    CONFIG['fp16']['mode'] = AMP_TYPE.TORCH
-    gpc._config = Config(CONFIG)
-    run_train()
-
-
-def run_with_apex_amp():
-    # hack config
-    CONFIG['fp16']['mode'] = AMP_TYPE.APEX
-    gpc._config = Config(CONFIG)
-    run_train()
-
-
-def run_with_naive_amp():
-    # hack config
-    CONFIG['fp16']['mode'] = AMP_TYPE.NAIVE
-    gpc._config = Config(CONFIG)
-    run_train()
+    get_components_func = non_distributed_component_funcs.get_callable(model_name)
+    gpc.config.fp16['mode'] = amp_mode
+    model_builder, train_dataloader, _, optimizer_class, criterion = get_components_func()
+
+    model = model_builder(checkpoint=False)
+    engine, train_dataloader, *args = colossalai.initialize(model=model,
+                                                            optimizer=optimizer_class(model.parameters(), lr=1e-3),
+                                                            criterion=criterion,
+                                                            train_dataloader=train_dataloader)
+
+    try:
+        engine.train()
+        for data, label in train_dataloader:
+            engine.zero_grad()
+            data = data.cuda()
+            label = label.cuda()
+            if criterion:
+                output = engine(data)
+                loss = engine.criterion(output, label)
+            else:
+                loss = engine(data, label)
+            engine.backward(loss)
+            engine.step()
+            break
```

Comment:
The unittest just check if it can run smoothly?

Reply:
Yes.